src/VirtualFileSystem/Wrapper/FileHandler.php

```diff
@@ -149,7 +149,7 @@ public function atEof()
     public function truncate($newSize = 0)
     {
         $this->position(0);
-        $newData = substr($this->file->data(), 0, $newSize);
+        $newData = substr($this->file->data() ? $this->file->data() : '', 0, $newSize);
```

Comment:
I can confirm that this fixes the issue. Maybe we could use null coalescing operator?